src/h5web/providers/hsds/api.ts

```diff
@@ -178,83 +143,121 @@ export class HsdsApi implements ProviderAPI {
     return Promise.all(attrsPromises);
   }
 
-  /* Processing methods to fetch links and attributes in addition to the entity. Also updates API members. */
-
   private async processGroup(
```

Comment:
All the `process` methods now return `Entity` objects (and objects that inherit `Entity`) directly. We don't go through intermediary `HDF5Dataset/Datatype/Group` objects any more and the `buildTree/Entity/...` utilities are obsolete!